code/asteroid/asteroid.cpp

```diff
@@ -796,35 +806,13 @@ void asteroid_create_debris_field(int num_asteroids, int asteroid_speed, int deb
 	Asteroid_field.speed = (float)asteroid_speed;
 	Asteroid_field.debris_genre = DG_DEBRIS;
 
-	for (int j = 0; j < MAX_ACTIVE_DEBRIS_TYPES; j++) {
-		Asteroid_field.field_debris_type[j] = -1;
-	}
+	Asteroid_field.field_debris_type.clear();
 
-	for (int j = 0; j < NUM_ASTEROID_SIZES; j++) {
-		Asteroid_field.field_asteroid_type[j] = false;
-	}
+	Asteroid_field.field_asteroid_type.clear();
 
-	int count = 0;
-	for (int i = 0; i < MAX_ACTIVE_DEBRIS_TYPES; i++) {
-		if (debris1 >= 0) {
-			Asteroid_field.field_debris_type[i] = debris1;
-			debris1 = -1;
-			count++;
-			continue;
-		}
-		if (debris2 >= 0) {
-			Asteroid_field.field_debris_type[i] = debris2;
-			debris2 = -1;
-			count++;
-			continue;
-		}
-		if (debris3 >= 0) {
-			Asteroid_field.field_debris_type[i] = debris3;
-			debris3 = -1;
-			count++;
-			continue;
-		}
-	}
+	Asteroid_field.field_debris_type = debris_types;
+
+	int count = static_cast<int>(Asteroid_field.field_debris_type.size());
 
 	Asteroid_field.num_used_field_debris_types = count;
```

Comment:
Since we're already refactoring here, we should just remove `num_used_field_debris_types` entirely and use `Asteroid_field.field_debris_type.size()` in it's place everywhere it's used. Less opportunity for these variables to go out of sync and break stuff...

Reply:
Took care of this. There is one spot it was used for "backwards compatibility" that seems less than ideal. You'll see my comments in the code. I'd like your opinion.

Reply:
I'd agree on less then ideal, but it's fine for now. Probably indeed hardcoded for retail or something.
If it turns out that this isn't needed anymore at some point, we can still remove it later.